pkg/cmd/cli/cmd/deploy.go

```diff
@@ -218,16 +246,58 @@ func (c *retryDeploymentCommand) retry(config *deployapi.DeploymentConfig, out i
 	return err
 }
 
+// cancelDeploymentCommand cancels the in-progress deployments.
+type cancelDeploymentCommand struct {
+	client deployCommandClient
+}
+
+// cancel cancels any deployment process in progress for config.
+func (c *cancelDeploymentCommand) cancel(config *deployapi.DeploymentConfig, out io.Writer) error {
+	deployments, err := c.client.ListDeploymentsForConfig(config.Namespace, config.Name)
+	if err != nil {
+		return err
+	}
+	for _, deployment := range deployments.Items {
+		status := deployutil.DeploymentStatusFor(&deployment)
+
+		switch status {
+		case deployapi.DeploymentStatusNew,
+			deployapi.DeploymentStatusPending,
+			deployapi.DeploymentStatusRunning:
+
+			if deployutil.IsDeploymentCancelled(&deployment) {
+				return nil
+			}
+
+			deployment.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
+			deployment.Annotations[deployapi.DeploymentStatusReasonAnnotation] = deployapi.DeploymentCancelledByUser
+			_, err := c.client.UpdateDeployment(&deployment)
+			if err == nil {
+				fmt.Fprintf(out, "cancelled #%d\n", config.LatestVersion)
+			} else {
+				return fmt.Errorf("couldn't cancel deployment %s (status: %s): %v", deployutil.DeploymentVersionFor(&deployment), status, err)
```

Comment:
Would it make sense to print the error here and keep iterating on the deployments, instead of returning here? Like you do when a cancellation succeeds.